lib/hocon/impl/tokenizer.rb

```diff
@@ -280,7 +374,7 @@ def pull_quoted_string
       while c != '"'
         c = next_char_raw
         if c == -1
-          raise self.problem(@line_origin, c, "End of input but string quote was still open", false, nil)
+          raise TokenIterator.problem(@line_origin, c, "End of input but string quote was still open", false, nil)
```

Comment:
That sounds reasonable, because then if it was being called from a subclass it would reference self's class instead of the parent class, I think